src/Mvc/Mvc.Razor/src/Diagnostics/MvcDiagnostics.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Razor;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Microsoft.AspNetCore.Mvc.Diagnostics
+{
+    public sealed class BeforeViewPage : MvcDiagnostic
+    {
+        public const string EventName = EventNamespace + 
+            "Razor." +
```

Comment:
Since we are changing some event names; though these ones have the additional "Razor." in their name?

Reply:
Wait which ones change? I don't think we want to change any event names. I made a comment before about the **class names**. Sorry if that was unclear.

Reply:
https://github.com/aspnet/AspNetCore/pull/11730#discussion_r302622142 event names are `nameof(classname)` so the current event name drives the class name?

Reply:
https://github.com/aspnet/AspNetCore/pull/11730#discussion_r302622142 event names are `nameof(classname)` so the current event name drives the class name?

Weird if there is a mis-match; that you listen to one name and get different named events?

Reply:
That's just the code you wanted to write 😆. I don't think it's a good idea for us to just change the event names and break people. 

I don't think it's weird if the class names add clarity, since the event names are now constants, you don't really see them.

The reason I brought this up is because of events like the following:

```txt
M.A.Mvc.BeforeAction (Before we begin processing for an action)
M.A.Mvc.BeforeOnActionExecution (Before an async action filter)
```

These seem like they are interchangeable but they are super different.

if you really feel like this naming thing is too much, just leave the names as they were originally and I'll do a pass after this is merged.

Reply:
I'll let you pick them up as I think you know what you want them named more :)